internal/pkg/service/buffer/sink/tablesink/storage/repository/slice_repo.go

```diff
@@ -48,7 +48,7 @@ func (r *SliceRepository) Get(k storage.SliceKey) op.ForType[*op.KeyValueT[stora
 		})
 }
 
-func (r *SliceRepository) Create(fileKey storage.FileKey, volumeID storage.VolumeID) *op.AtomicOp[storage.Slice] {
+func (r *SliceRepository) Create(fileKey storage.FileKey, volumeID storage.VolumeID, prevSliceSize datasize.ByteSize) *op.AtomicOp[storage.Slice] {
```

Comment:
Unrelated minor fix: there was missing `prevSliceSize` in the top-level repository method.